### PR TITLE
[WIP] :bird: :punch: promise

### PR DESCRIPTION
**Pull request by iamdustan**
Handles my current environment that resulted in #64. There is probably a better test for this than what I have done though...

### CODE REVIEW

File: src/Resolver.js [hunk 1]
```diff
@@ -159,7 +159,8 @@ export default class Resolver extends React.Component {
         const factory = resolve[name];
         const value = factory(props);
 
-        if (value instanceof Promise) {
+        // duckpunched promise
+        if (value instanceof Promise || typeof value.then === 'function') {
```
Comment (ericclemmons) [inline src/Resolver.js]: Can you switch to double-quotes to match to match the rest of the coding style?  (I guess I forgot to `.eslintrc` this)

**Comment by ericclemmons** [comment]
@iamdustan What library are you using for fetching?  I want to test out a few to ensure this works with `Promise.all` on the server.

**Comment by ericclemmons** [comment]
Related to #64.

File: src/Resolver.js (hunk 1 above)
Comment (iamdustan) [inline src/Resolver.js]: eslint was eating it for me. ran `eslint src` (instead of `npm run lint`) to get the error. All updated now.

**Comment by iamdustan** [comment]
I am using https://github.com/matthew-andrews/isomorphic-fetch which requires an external Promise implementation. I likely am experiencing some issues between `require('es6-promise').polyfill()`, babel’s core-js rewriting, and multiple node modules which are also compiled with babel.

**Comment by ericclemmons** [comment]
I was talking to people about this today & they recommended wrapping pretty much everything in `Promise.resolve`, which is supposed to solve the duck-punch problem for us?

Have you heard about this?  As in, `Promise` **has** to exist for this project to run.

**Comment by ericclemmons** [comment]
Hah!  I got bit by this bug that's actually an issue with the compiled code, I'd wager!
If I `@resolve("multilead", () => "Howdy")`, it's fine.

If I `@resolve("multilead", () => Promise.resolve("Howdy"))`, I get:

```
{ 'Symbol(record)_1.ael8nu52kh9kvs4i': 
   { p: [Circular],
     c: [],
     a: undefined,
     s: 1,
     d: true,
     v: 'Howdy',
     h: false,
     n: true } }
```

**Comment by ericclemmons** [comment]
This needed a bit more finagling, but this works:

```js
          if (value && typeof value.then === "function") {
            nextState.pending[name] = value;
          } else {
```

I'd like to combine the two together for clarity.  I should have this resolved tomorrow.

**Comment by ericclemmons** [comment]
@iamdustan Thanks for this.  This was huge, especially considering there are major issues with `instanceof Promise` between Babel-ified projects.

**Comment by iamdustan** [comment]
:raised_hands: woohoo!

**Pull request merged**